src/app-layer-parser.c

```diff
@@ -1468,9 +1462,9 @@ int AppLayerParserParse(ThreadVars *tv, AppLayerParserThreadCtx *alp_tctx, Flow
         AppLayerIncTxCounter(tv, f, cur_tx_cnt - p_tx_cnt);
     }
 
-    /* stream truncated, inform app layer */
+    /* stream truncated, trigger raw stream reassembly */
     if (flags & STREAM_DEPTH)
-        AppLayerParserStreamTruncated(pstate, f->proto, alproto, f->alstate, flags);
+        AppLayerParserTriggerRawStreamReassembly(f, direction);
```

Comment:
This looks strange to me, do we set another flag somewhere else when depth is reached ?

Reply:
Q: flag for which part of the code exactly? applayer?

My idea here was that whenever we encounter stream depth condition, we make whatever data is available until now in the raw stream to the applayer. This is what was done in truncate fn calls too which I have removed.

Reply:
Is this not already done by the detect/stream engine understanding the depth logic?

Reply:
Why are the `APP_LAYER_PARSER_TRUNC_TS` flags removed ?
Are some other flags equivalent ?

`ts_disrupt_flags` gets `STREAM_DEPTH` flag from `FlowGetDisruptionFlags` which gets used by `AppLayerParserGetStateProgress` with `IS_DISRUPTED`
No need for APP_LAYER_PARSER_TRUNC_TS flag.
That is what I want to see somehow in the commit message